src/spec/create_extension_spec.py

```diff
@@ -91,12 +88,20 @@ def main():
                 quantity="?",
             ),
         ],
+        attributes=[
+            NWBAttributeSpec(
+                name="description",
+                dtype="text",
+                doc="description",
+                default_value="A table describing the optical sources of the device."
+            )
+        ]
     )
 
     nirs_detectors = NWBGroupSpec(
         neurodata_type_def="NIRSDetectorsTable",
         neurodata_type_inc="DynamicTable",
-        name="detectors",
+        default_name="NIRSDetectorsTable",
         doc="A table describing optical detectors of a NIRS device",
```

Comment:
Does the `doc` field for a spec have any impact on a class auto-built via `get_class`? Or is this just used when generating the docs for the schema (eg. https://nwb-schema.readthedocs.io/en/latest/format.html#sec-timeintervals)?

Reply:
I know it is written to the schema and is a required field. It might be used as the docstring of the auto-generated class- I can't remember.